service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -221,9 +221,12 @@ private String buildExplanationPertinentInformation(Immunization immunization) {
             explanation = CodeableConceptMappingUtils.extractTextOrCoding(reason);
             return explanation.map(value -> REASON + value).orElse(StringUtils.EMPTY);
         } else if (immunization.hasExplanation() && immunization.getExplanation().hasReasonNotGiven()) {
-            CodeableConcept reasonNotGiven = immunization.getExplanation().getReasonNotGivenFirstRep();
-            explanation = CodeableConceptMappingUtils.extractTextOrCoding(reasonNotGiven);
-            return explanation.map(value -> REASON_NOT_GIVEN + value).orElse(StringUtils.EMPTY);
+            String reasonNotGiven = immunization.getExplanation().getReasonNotGiven().stream()
+                .map(CodeableConceptMappingUtils::extractTextOrCoding)
+                .map(Optional::get)
+                .filter(value -> !value.equals(StringUtils.EMPTY))
```

Comment:
This will throw exception if value is null. It's better to invert the check `!StringUtils.EMPTY.equals(value)` and even better, use Apache helper method `.filter(StringUtils::isNotBlank)` (included in my code example above)